### PR TITLE
Add Instruction cbegin and cend to access operands

### PR DESCRIPTION
**Pull request by dneto0**

### CODE REVIEW

**Comment by dneto0** [comment]
This would be useful for some spots in https://github.com/KhronosGroup/SPIRV-Tools/pull/324 
where you just want to iterate over operands but don't need their index at the same time.  So you can just do a range-based-for.

Looking for @antiagainst review

File: source/opt/instruction.h [hunk 1]
```diff
@@ -117,6 +119,10 @@ class Instruction {
     return dbg_line_insts_;
   }
 
+  // Const begin and end iterators for operands.
+  const_iterator cbegin() { return operands_.cbegin(); }
```
Comment (antiagainst) [inline source/opt/instruction.h]: `const_iterator ...() const`?

File: test/opt/test_instruction.cpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+TEST(InstructionTest, CreateWithOpcodeAndOperands) {
+  auto i32 = MakeInstruction(SpvOpTypeInt, {44, 32, 1});
+  spv_parsed_operand_t parsed_operands[] = {
+      {1, 1, SPV_OPERAND_TYPE_RESULT_ID, SPV_NUMBER_NONE, 0},
+      {2, 1, SPV_OPERAND_TYPE_LITERAL_INTEGER, SPV_NUMBER_UNSIGNED_INT, 32},
+      {3, 1, SPV_OPERAND_TYPE_LITERAL_INTEGER, SPV_NUMBER_UNSIGNED_INT, 1},
+  };
+  spv_parsed_instruction_t parsed_inst{i32.data(),
+                                       uint16_t(i32.size()),
```
Comment (antiagainst) [inline test/opt/test_instruction.cpp]: Wow... `clang-format` sometime really does a "nice" job.

But I won't complain anyway.

File: test/opt/test_instruction.cpp [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  EXPECT_EQ(3u, inst.NumOperandWords());
+  EXPECT_EQ(2u, inst.NumInOperandWords());
+  auto cbegin = inst.cbegin();
+  auto cend = inst.cend();
+  EXPECT_NE(cend, inst.cbegin());
+
+  // Spot check iteration across operands.
+
+  auto iter = inst.cbegin();
+  const auto& operand0 = *iter;
```
Comment (antiagainst) [inline test/opt/test_instruction.cpp]: It seems we can do something like
```
for (int i = 0; i < 3; ++i) {
  const auto& operand = *iter;
  EXPECT_THAT(operand.type, Eq(parsed_operands[i]..));
  EXPECT_THAT(operand.words, Eq(std::vector<uint32_t>{i32[i + 1]}));
  EXPECT_NE(cend, iter++);
}
```

Comment (dneto0) [inline test/opt/test_instruction.cpp]: Some advice tells you not to have too much flow control in your tests.  I rewrote it your way anyway.

File: source/opt/instruction.h (hunk 1 above)
Comment (dneto0) [inline source/opt/instruction.h]: Thanks!  Good catch.

**Comment by dneto0** [comment]
Rebased

**Comment by antiagainst** [comment]
LGTM

**Pull request merged**